app/lib/efile/md/md502_r_calculator.rb

```diff
@@ -61,16 +61,16 @@ def calculate_line_8
       def calculate_line_9a
         if @intake.direct_file_data.fed_ssb.positive?
           if @intake.filing_status_mfj?
-            @intake.primary_ssb_amount&.round || 0
+            @intake.direct_file_json_data.primary_filer_social_security_benefit_amount&.round || 0
           else
             @intake.direct_file_data.fed_ssb.round
           end
         end
       end
 
       def calculate_line_9b
-        if @intake.filing_status_mfj? && @intake.direct_file_data.fed_ssb.positive? && @intake.spouse_ssb_amount.present?
-          @intake.spouse_ssb_amount.round
+        if @intake.filing_status_mfj? && @intake.direct_file_data.fed_ssb.positive?
+          @intake.direct_file_json_data.spouse_filer_social_security_benefit_amount&.round
```

Comment:
just noticing that `calculate_line_9a` returns `0` if the amount is nil` but `calculate_line_9b` returns `nil` if it's nil, probably should choose one or the other, although it'll end up in the same place since this ultimately uses `add_element_if_present`

Reply:
Changed so that they both return nil if no value is present